modules/nextflow/src/main/groovy/nextflow/trace/AnsiLogObserver.groovy

```diff
@@ -371,34 +388,78 @@ class AnsiLogObserver implements TraceObserver {
     protected String fmtChop(String str, int cols) {
         if( str.size() <= cols )
             return str
-        return cols>3 ? str[0..(cols-3-1)] + '...' : str[0..cols-1]
+        return cols>5 ? str.take(3) + '…' + str.takeRight(cols-1-3) : str[0..cols-1]
     }
 
-    protected String line(ProgressRecord stats) {
+    private final static Pattern TAG_REGEX = ~/ \((.+)\)( *)$/
+    private final static Pattern LBL_REPLACE = ~/ \(.+\) *$/
+
+    protected Ansi line(ProgressRecord stats) {
+        final term = ansi()
         final float tot = stats.getTotalCount()
         final float com = stats.getCompletedCount()
         final label = fmtWidth(stats.taskName, labelWidth, Math.max(cols-50, 5))
+        final tagMatch = TAG_REGEX.matcher(label)
```

Comment:
Inline regex e.g. `~/something/` requires to "compile" the pattern every time. A static variable should be a bit more efficient